ui/plugin-system/src/components/PanelSpecEditor/PanelSpecEditor.tsx

```diff
@@ -44,31 +44,21 @@ export function PanelSpecEditor(props: PanelSpecEditorProps) {
     throw new Error(`Missing implementation for panel plugin with kind '${kind}'`);
   }
 
-  const getQueryType = (): string => {
+  const getQueryTypes = (): string[] => {
+    if (plugin.supportedQueryInterfaces && plugin.supportedQueryInterfaces.length > 0) {
+      return plugin.supportedQueryInterfaces;
+    }
     const queriesList = panelDefinition?.spec?.queries;
     if (queriesList === undefined) {
-      return '';
+      return [''];
```

Comment:
Wondering why do we need return an array with a empty item 👀

Reply:
Not sure if we need. I'll check later.